pkg/kubelet/apis/cri/v1alpha1/api.proto

```diff
@@ -971,33 +977,75 @@ message UInt64Value {
     uint64 value = 1;
 }
 
-// FsInfo contains data about filesystem usage.
-message FsInfo {
-    // The block device name associated with the filesystem.
-    string device = 1;
-    // The root directory for the images.
-    string path = 2;
-    // CapacityBytes represents the total capacity (bytes) of the filesystems
-    // underlying storage.
-    UInt64Value capacity_bytes = 3;
-    // AvailableBytes represents the storage space available (bytes) for the
-    // filesystem.
-    UInt64Value available_bytes = 4;
+// StorageIdentifier uniquely identify the storage..
+message StorageIdentifier{
+    // UUID of the device.
+    string uuid = 1;
+}
+
+// FilesystemUsage provides the filesystem usage information.
+message FilesystemUsage {
+    // Timestamp in nanoseconds at which the information were collected. Must be > 0.
+    int64 timestamp = 1;
+    // The underlying storage of the filesystem.
+    StorageIdentifier storage_id = 2;
     // UsedBytes represents the bytes used for images on the filesystem.
     // This may differ from the total bytes used on the filesystem and may not 
     // equal CapacityBytes - AvailableBytes.
-    UInt64Value used_bytes = 5;
-    // InodesCapacity represents the total inodes in the filesystem.
-    UInt64Value inodes_capacity = 6;
-    // InodesAvailable represents the free inodes in the filesystem.
-    UInt64Value inodes_available = 7;
+    UInt64Value used_bytes = 3;
     // InodesUsed represents the inodes used by the images.
     // This may not equal InodesCapacity - InodesAvailable because the underlying
     // filesystem may also be used for purposes other than storing images.
-    UInt64Value inodes_used = 8;
+    UInt64Value inodes_used = 4;
 }
 
 message ImageFsInfoResponse {
     // filesystem information of images.
-    FsInfo fs_info = 1;
+    FilesystemUsage image_fs = 1;
+}
+
+message ContainerStatsRequest{
+    // ID of the container for which to retrieve stats.
+    string container_id = 1;
+}
+
+message ContainerStatsResponse {
+    // Stats of the container.
+    ContainerStats stats = 1;
+}
+
+message ListContainerStatsRequest{
+}
+
+message ListContainerStatsResponse {
+    // Stats of the container.
+    repeated ContainerStats stats = 1;
+}
+
+// ContainerStats provides the resource usage statistics for a container.
+message ContainerStats {
+    // Information of the container.
+    Container container = 1;
```

Comment:
Labels and annotations are needed for kubelet to recognize the pod-level information associated with the container. We can
 1. include all necessary information: metadata, id, labels, annotations,
 2. similar to (1), but encapsulate all the information in a new type that's similar to k8s API's metadata.
 3. include the `Container`

I chose (3) because (1) seems unnecessarily tedious. I could do (2) but it involves more API changes, and may break some implementations.
@Random-Liu @mrunalp, any preferences?